geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/view/RouteSelectorByAttribute.scala

```diff
@@ -0,0 +1,69 @@
+/***********************************************************************
+ * Copyright (c) 2013-2019 Commonwealth Computer Research, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, Version 2.0
+ * which accompanies this distribution and is available at
+ * http://www.opensource.org/licenses/apache2.0.php.
+ ***********************************************************************/
+
+package org.locationtech.geomesa.index.view
+
+import org.geotools.data.{DataStore, Query}
+import org.locationtech.geomesa.filter.FilterHelper
+import org.opengis.feature.simple.SimpleFeatureType
+
+import scala.util.control.NonFatal
+
+/**
+  * Selects routes based on the attributes present in the query filter
+  */
+class RouteSelectorByAttribute extends RouteSelector {
+
+  import RouteSelectorByAttribute.RouteAttributes
+
+  import scala.collection.JavaConverters._
+
+  private var mappings: Seq[(Set[String], DataStore)] = Seq.empty
+  private var id: Option[DataStore] = None
+  private var include: Option[DataStore] = None
+
+  override def init(stores: Seq[(DataStore, java.util.Map[String, AnyRef])]): Unit = {
+    val builder = Seq.newBuilder[(Set[String], DataStore)]
+    stores.foreach { case (ds, config) =>
+      Option(config.get(RouteAttributes)).foreach { attributes =>
+        try {
+          attributes.asInstanceOf[java.util.List[AnyRef]].asScala.foreach {
+            case s: String if s.equalsIgnoreCase("id") => id = Some(ds)
+            case s: String => builder += Set(s) -> ds
+            case list: java.util.List[String] if list.isEmpty => include = Some(ds)
```

Comment:
If I'm reading this right, in the case of multiple stores being configured for Filter.INCLUDE, only the last is used?

Reply:
yeah, that would be an error case - I can add a check for that

Reply:
fixed